src/network/server.rs

```diff
@@ -90,15 +100,50 @@ impl<N: Network, E: Environment> Server<N, E> {
         let prover = Prover::open::<RocksDB, _>(
             &mut tasks,
             &prover_storage_path,
-            miner,
+            address,
             local_ip,
+            pool_ip,
             &status,
             &terminator,
             peers.router(),
             ledger.reader(),
             ledger.router(),
         )
         .await?;
+        // Initialize a new instance for managing the operator.
+        let operator = Operator::open::<RocksDB, _>(
+            &mut tasks,
+            &operator_storage_path,
+            address,
+            local_ip,
+            prover.memory_pool(),
+            peers.router(),
+            ledger.reader(),
+            ledger.router(),
+            prover.router(),
+        )
+        .await?;
+
+        // TODO (howardwu): This is a hack for the prover.
```

Comment:
is the hack that possibly an extra connection is attempted instead of checking for an existing connection first?

Reply:
@howardwu